Zend/zend_vm_def.h

```diff
@@ -5554,7 +5554,112 @@ ZEND_VM_C_LABEL(num_index):
 	ZEND_VM_NEXT_OPCODE_CHECK_EXCEPTION();
 }
 
-ZEND_VM_HANDLER(71, ZEND_INIT_ARRAY, CONST|TMP|VAR|CV, CONST|TMPVAR|UNUSED|NEXT|CV, ARRAY_INIT|REF)
+ZEND_VM_HANDLER(208, ZEND_ADD_ARRAY_UNPACK, ANY, ANY)
+{
+	USE_OPLINE
+	zend_free_op free_op1;
+	zval *op1;
+	
+	SAVE_OPLINE();
+	op1 = GET_OP1_ZVAL_PTR(BP_VAR_R);
+	
+	SEPARATE_ARRAY(EX_VAR(opline->result.var));
```

Comment:
Instead of separating here, I'd suggest to replace the ZVAL_EMPTY_ARRAY in INIT_ARRAY with an explicit (non-interned) array creation.